runtime/src/utils.rs

```diff
@@ -322,6 +323,27 @@ impl Signaler {
     }
 }
 
+/// Creates a Rayon thread pool that uses the runtime's `spawn_blocking` to spawn threads.
+///
+/// # Arguments
+/// - `context`: The runtime context implementing the `Spawner` trait.
+/// - `concurrency`: The number of tasks to spawn in the pool.
+///
+/// # Returns
+/// A `Result` containing the configured `ThreadPool` or an error if the pool cannot be built.
+pub fn create_rayon_pool<S: Spawner>(
```

Comment:
I suspect this will be useful for many data-processing patterns (nice that we can build this over tokio)